ql/src/java/org/apache/hadoop/hive/ql/parse/RowResolver.java

```diff
@@ -125,6 +125,22 @@ private boolean putInternal(String tabAlias, String colAlias, ColumnInfo colInfo
     return false;
   }
 
+  /**
+   * Puts a resolver entry corresponding to a aggregate expression and its
+   * column alias which is used for alias recognition in the having clause.
+   * @param exprToColumnAlias
+   *          The map containing the expression to alias mapping
+   * @return
+   * @throws SemanticException
+   */
+  public void putAggregateAlias(Map<ASTNode, String> exprToColumnAlias) throws SemanticException {
```

Comment:
The name of the method is unnecessarily tight to aggregates and aliases while the implementation seems much more generic than that. Given that we just call `put` for every entry in the map maybe a better name would be `putAll` or maybe `replaceAll` since we only perform the action if an entry is already there.

Will this logic really add an entry in the resolver or it just changes the alias for ColumnInfo that is registered previously? Depending on the answer another potential name would be `replaceAliases`.

A well chosen name will better indicate what the method really does without requiring checking the implementation.